src/libraries/System.Diagnostics.Process/tests/ProcessModuleTests.cs

```diff
@@ -93,9 +93,10 @@ public void ModulesAreDisposedWhenProcessIsDisposed()
         }
 
         public static bool Is_LongModuleFileNamesAreSupported_TestEnabled
-            => PathFeatures.AreAllLongPathsAvailable() // we want to test long paths
+            => OperatingSystem.IsWindows() // it's specific to Windows
+            && PathFeatures.AreAllLongPathsAvailable() // we want to test long paths
             && !PlatformDetection.IsMonoRuntime // Assembly.LoadFile used the way this test is implemented fails on Mono
-            && OperatingSystem.IsWindowsVersionAtLeast(8); // it's specific to Windows and does not work on Windows 7
+            && PlatformDetection.Is64BitProcess; // for some reason it's flaky on x86
```

Comment:
@JeremyKuhne do you perhaps know about any long path issues specific to x86?

Reply:
can you explain what do you mean by flaky? what's happening? I'd expect this to be deterministic

Reply:
should this be `!(Is32Bit && !IsArm)` - perhaps we should add IsX86 in the PlatformDetection - we should also open the issue to try to understand why this is failing on x86 I think (unless we don't care for some reason)

Reply:
My guess is that it's rather related to assembly loading on x86 or process modules information not being refreshed rather than long paths itself.

> I'd expect this to be deterministic

It's not ;( All tests have passed in #57335 where I've re-enabled this test but it fails from time to time: #57452.

Reply:
Perhaps we should add more diagnostics.. I.e. print all module list, enumerate modules in the assembly returned by Assembly.LoadFile. This sound rather sketchy as is - perhaps it's unrelated issue completely but we should at least understand why it's happening before we disable I think

Reply:
> RuntimeInformation.ProcessArchitecture != Architecture.X86

done

Reply:
> @JeremyKuhne do you perhaps know about any long path issues specific to x86?

The only issues I've ever seen are bugs with the WOW layer when running an x86 binary on x64. I'm unaware of any outstanding issues though.